lib/testutils/httpmultibin.go

```diff
@@ -116,12 +127,48 @@ func getWebsocketCloserHandler(t testing.TB) http.Handler {
 	})
 }
 
+func writeJSON(w io.Writer, v interface{}) error {
+	e := json.NewEncoder(w)
+	e.SetIndent("", "  ")
+	return errors.Wrap(e.Encode(v), "failed to encode JSON")
+}
+
+func getEncodedHandler(t testing.TB, compressionType httpext.CompressionType) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		var (
+			encoding string
+			err      error
+			encw     io.WriteCloser
+		)
+
+		switch compressionType {
+		case httpext.CompressionTypeBr:
+			encw = brotli.NewWriter(rw)
+			encoding = "br"
+		case httpext.CompressionTypeZstd:
+			encw, _ = zstd.NewWriter(rw)
+			encoding = "zstd"
+		}
+
+		rw.Header().Set("Content-Type", "application/json")
+		rw.Header().Add("Content-Encoding", encoding)
+		data := jsonBody{Compression: encoding}
```

Comment:
Nice catch! I was considering replicating the full httpbin.org response to match the other endpoints, but ultimately decided against it since it's not used in the tests.

I now copied the request headers, as you mentioned.